### ISSUE TITLE
Shared memory not freed properly

### ISSUE THREAD

**Issue by generalbytes** [pcdv/jocket#5]
When the following code is run it incrementally allocates memory until it takes full capacity of /run/shm (/dev/shm) despite the fact that all of the close() methods on sockets are called.
**_Is there a way how to correctly free JocketSocket/JocketFile so it doesn't occupy any more space in shm?_**

```java
package com.generalbytes.bus.test;

import jocket.net.JocketSocket;
import jocket.net.ServerJocket;

import java.io.IOException;

public class TestX {
    public static void main(String[] args) {
        try {
            final ServerJocket ssocket = new ServerJocket(2000,1024,134217728);
            new Thread(new Runnable() {
                @Override
                public void run() {
                    while (true) {
                        try {
                            final JocketSocket accept = ssocket.accept();
                            if (accept == null) {
                                return;
                            }else{
                                accept.close();
                            }
                        } catch (IOException e) {
                            e.printStackTrace();
                        } catch (Exception e) {
                            e.printStackTrace();
                        }
                    }
                }
            }).start();
            try {
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            for (int i=0;i<10000;i++){
                    JocketSocket sock = new JocketSocket(2000);
                    sock.close();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
```

**Comment by generalbytes** [comment]
After more investigation the issue seems to be that mapped buffer in JocketFile is freed only when JVM is stopped.
Adding following code into JocketFile and calling it from JocketSocket fixes the issue:
  private static void unmap(MappedByteBuffer buffer)
  {
    sun.misc.Cleaner cleaner = ((DirectBuffer) buffer).cleaner();
    cleaner.clean();
  }

  public void freeBuffer() {
    unmap(buf);
  }

**Comment by pcdv** [comment]
Thanks for spotting the issue. I reproduced it and integrated your fix (cbd5749). This will probably not work with Java9 but that's probably not the only thing that will break :)
